sycophancy_analysis/dashboard/app.py

```
@@ -0,0 +1,928 @@
+    except Exception:
+        prefix = None
+    if prefix:
+        # Accept both absolute and relative paths; restrict to existing dir within RESULTS_DIR subtree
+        try:
+            p = Path(prefix)
+            # If relative, resolve against BASE_DIR
+            if not p.is_absolute():
+                p = (BASE_DIR / p).resolve()
+            # Security: ensure the path is within RESULTS_DIR subtree to prevent path traversal
+            results_dir = RESULTS_DIR.resolve()
+            if p.exists() and p.is_dir() and results_dir in p.parents or p == results_dir:
```

Ruff (0.13.1)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)